sdk/keyvault/azure-security-keyvault/azure/security/keyvault/keys/README.md

```diff
@@ -0,0 +1,255 @@
+
+### Key Client:
+The Key client performs the interactions with the Azure Key Vault service for getting, setting, updating, deleting, and listing keys and its versions. An asynchronous and synchronous, KeyClient, client exists in the SDK allowing for selection of a client based on an application's use case. Once you've initialized a Key, you can interact with the primary resource types in Key Vault.
+
+## Examples
+The following section provides several code snippets using the above created `key_client`, covering some of the most common Azure Key Vault Key service related tasks, including:
+* [Create a Key](#create-a-key)
+* [Retrieve a Key](#retrieve-a-key)
+* [Update an existing Key](#update-an-existing-key)
+* [Delete a Key](#delete-a-key)
+* [List Keys](#list-keys)
+* [Async create a Key](#async-create-a-key)
+* [Async list Keys](#async-list-keys)
+
+### Create a Key
+`create_key` creates a Key to be stored in the Azure Key Vault. If a key with the same name already exists, then a new version of the key is created.
+```python
+
+# Create a key of any type
+key = key_client.create_key("key-name", "RSA-HSM")
+
+# Create an RSA key with size specification (optional)
+rsa_key = key_client.create_rsa_key("rsa-key-name", hsm=False, size=2048)
+
+# Create an EC key with curve specification and using HSM
+ec_key = key_client.create_key("ec-key-name", hsm=True)
+
+print(key.name)
+print(key.key_material.kty)
+
+print(rsa_key.name)
+print(rsa_key.key_material.kty)
+
+print(ec_key.name)
+print(ec_key.key_material.kty)
+```
+
+### Retrieve a Key
+`get_key` retrieves a key previously stored in the Key Vault.
+```python
+key = key_client.get_key("key-name")
+
+print(key.name)
+print(key.value)
+```
+
+### Update an existing Key
+`update_key` updates a key previously stored in the Key Vault.
+```python
+# Clients may specify additional application-specific metadata in the form of tags.
+tags = {"foo": "updated tag"}
+
+updated_key = key_client.update_key("key-name", tags=tags)
+
+print(updated_key.name)
+print(updated_key.version)
+print(updated_key.updated)
+print(updated_key.tags)
+
+```
+
+### Delete a Key
+`delete_key` deletes a key previously stored in the Key Vault. When [soft-delete](https://docs.microsoft.com/en-us/azure/key-vault/key-vault-ovw-soft-delete) is not enabled for the Key Vault, this operation permanently deletes the key.
+```python
+deleted_key = key_client.delete_key("key-name")
+
+print(deleted_key.name)
+print(deleted_key.deleted_date)
+```
+### List keys
+This example lists all the keys in the specified Key Vault.
+```python
+keys = key_client.list_keys()
+
+for key in keys:
+    # the list doesn't include values or versions of the keys
+    print(key.name)
+```
+
+### Async operations
+Python’s [asyncio package](https://pypi.org/project/asyncio/) (introduced in Python 3.4) and its two keywords `async` and `await` serves to declare, build, execute, and manage asynchronous code.
```

Comment:
asyncio entered the standard library in 3.4 but async/await were added in 3.5. We shouldn't mention 3.4 here at all, it sets someone up to believe we support it (our minimum 3.x is 3.5.3).